build/scripts/restore.ps1

```diff
@@ -0,0 +1,26 @@
+param ([string]$nugetZipUrl = $(throw "Need an URL to the NuGet zip") )
+
+$destination = ${env:UserProfile}
+$outFilePath = [IO.Path]::ChangeExtension([IO.Path]::GetTempFileName(), "zip")
+write-host "Downloading $nugetZipUrl -> $outFilePath"
+$client = new-object System.Net.WebClient
+$client.DownloadFile($nugetZipUrl, $outFilePath)
+
+# It's possible for restore to run in parallel on the test machines.  As such
+# we need to restore only new files to handle simultaneous restore scenarios.
```

Comment:
I don't really like this at all.  I thought @agocke added a restore semaphore to the max/linux builds.  Why not do the same here?